examples/classification/plot_classification_probability.py

```diff
@@ -71,25 +68,29 @@
     accuracy = accuracy_score(y, y_pred)
     print("Accuracy (train) for %s: %0.1f%% " % (name, accuracy * 100))
 
-    # View probabilities:
-    probas = classifier.predict_proba(Xfull)
-    n_classes = np.unique(y_pred).size
-    for k in range(n_classes):
-        plt.subplot(n_classifiers, n_classes, index * n_classes + k + 1)
-        plt.title("Class %d" % k)
-        if k == 0:
-            plt.ylabel(name)
-        imshow_handle = plt.imshow(
-            probas[:, k].reshape((100, 100)), extent=(3, 9, 1, 5), origin="lower"
+    for k in classifier.classes_:
+        disp = DecisionBoundaryDisplay.from_estimator(
+            classifier,
+            X,
+            plot_method="pcolormesh",
+            response_method="predict_proba",
+            class_label=k,
+            ax=axs[index, k],
+        )
+        axs[index, k].set(
+            xticks=(), yticks=(), ylabel=name if k == 0 else None, title=f"Class #{k}"
         )
-        plt.xticks(())
-        plt.yticks(())
         idx = y_pred == k
         if idx.any():
-            plt.scatter(X[idx, 0], X[idx, 1], marker="o", c="w", edgecolor="k")
-
-ax = plt.axes([0.15, 0.04, 0.7, 0.05])
-plt.title("Probability")
-plt.colorbar(imshow_handle, cax=ax, orientation="horizontal")
+            axs[index, k].scatter(
+                X[idx, 0], X[idx, 1], marker="o", c="w", edgecolor="k"
+            )
```

Comment:
I feel like the plot would be more informative if all data points are included, but colorcoded (or better shape coded as +, x, ...) so that the user can see why decision boundaries are as they are.